src/Mod/Path/PathScripts/PathAreaOp.py

```diff
@@ -538,6 +551,64 @@ def opExecute(self, obj, getsim=False): # pylint: disable=arguments-differ
         PathLog.debug("obj.Name: " + str(obj.Name) + "\n\n")
         return sims
 
+    # Template methods to be overwritten in operation sub-classes as needed
+    def initAreaOp(self, obj):
+        '''initAreaOp(obj) ... overwrite if the receiver class needs initialisation.
+        Can safely be overwritten by subclasses.'''
+        pass # pylint: disable=unnecessary-pass
+
+    def _initAreaOpAttributes(self, obj):
+        '''_initAreaOpAttributes(obj) ... overwrite to instantiate operation-specific
+        class instance attributes.
+        Can safely be overwritten by subclasses.'''
+        # pylint: disable=unused-argument
+        pass # pylint: disable=unnecessary-pass
+
+    def areaOpFeatures(self, obj):
+        '''areaOpFeatures(obj) ... overwrite to add operation specific features.
+        Can safely be overwritten by subclasses.'''
+        # pylint: disable=unused-argument
+        return 0
+
+    def areaOpOnChanged(self, obj, prop):
+        '''areaOpOnChanged(obj, porp) ... overwrite to process operation specific changes to properties.
+        Can safely be overwritten by subclasses.'''
+        pass # pylint: disable=unnecessary-pass
+
+    def areaOpOnDocumentRestored(self, obj):
+        '''areaOpOnDocumentRestored(obj) ... overwrite to fully restore receiver'''
+        pass # pylint: disable=unnecessary-pass
+
+    def areaOpPropertyDefinitions(self):
+        '''areaOpPropertyDefinitions() ... overwrite to provide
+        operation-specific property definitions.
+        Can safely be overwritten by subclasses.'''
+        return []
+
+    def areaOpPropertyEnumerations(self):
+        '''areaOpPropertyEnumerations() ... overwrite to provide
+        operation-specific enumeration lists for properties.
+        Can safely be overwritten by subclasses.'''
+        return {}
+
+    def areaOpPropertyDefaults(self, obj, job):
+        '''areaOpPropertyDefaults(obj, job) ...
+        This method should be overwritten by the operation sub-class.
+        Default property values should be assigned to class dictionary
+        attribute as self.propDefaults['property_name'] = value'''
+        return {}
+
+    def areaOpSetEditorModes(self, obj):
+        '''areaOpSetEditorModes(obj) ...
+        Process operation-specific changes to properties visibilities.
+        This method may be overwritten by subclasses.'''
+        pass
+
+    def areaOpSetDefaultValues(self, obj, job):
+        '''areaOpSetDefaultValues(obj, job) ... overwrite to set initial values of operation specific properties.
+        Can safely be overwritten by subclasses.'''
+        pass # pylint: disable=unnecessary-pass
```

Comment:
it doesn't look like we preserve or maintain `pylint` checking, might as well get rid of those.